static/js/components/dashboard/CourseAction.js

```diff
@@ -21,19 +22,27 @@ export default class CourseAction extends React.Component {
   makeEnrollButton = (text: string, run: CourseRun, disabled: boolean) => {
     const { checkout } = this.props;
     let onClick;
+    let button;
+
     if (!disabled) {
       onClick = () => {
         checkout(run.course_id);
       };
+      button = (
+        <Button type='button' className="dashboard-button" onClick={onClick}>
+          {text}
+        </Button>
+      );
+    } else {
+      button = (
+        <Button disabled type='button' className="dashboard-button">
+          {text}
+        </Button>
+      );
     }
+
     return <span>
-      <button
-        className="dashboard-button"
-        disabled={disabled}
-        onClick={onClick}
-      >
-        {text}
-      </button>
+      {button}
```

Comment:
i think this function can be rewritten like this:

```javascript
  makeEnrollButton = (text: string, run: CourseRun, disabled: boolean) => {
    const { checkout } = this.props;
    let buttonProps = {};

    if (!disabled) {
      buttonProps.onClick = () => {
        checkout(run.course_id);
      };
    } else {
      buttonProps.disabled = true;
    }

    return <span>
      <Button className="dashboard-button" {...buttonProps}>
        {text}
      </Button>
      <span className="sr-only"> in {run.title}</span>
    </span>;
  };
```

side note: i don't think the <Button /> component needs type='button'. not sure about that, though